distributed/deploy/spec.py

```diff
@@ -404,16 +404,15 @@ async def _close(self):
             if isawaitable(f):
                 await f
             await self._correct_state()
-            for future in self._futures:
-                await future
-            async with self._lock:
-                with suppress(CommClosedError, OSError):
-                    if self.scheduler_comm:
-                        await self.scheduler_comm.close(close_workers=True)
-                    else:
-                        logger.warning("Cluster closed without starting up")
+            await asyncio.gather(*self._futures)
+
+            if self.scheduler_comm:
+                await self.scheduler_comm.close_rpc()
```

Comment:
From how I read the code, `close` like
```python
self.scheduler_comm.close(close_workers=True)
```
can actually never work since there is no RPC handler called `close` but we'd never know because we suppress all OSErrors. 
I replaced it with the proper RPC handler, `terminate`

---

If we're not calling `self.scheduler_comm.close_rpc` we need to rely on the finalizer which is something I'd like to avoid.

Reply:
The latest change passes locally, if you think that solution is reasonable/appropriate then I'm good with that too, gpuCI tests already passed.